component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java

```diff
@@ -731,4 +834,57 @@ private void setUserClaimValues(AuthenticatedUser authenticatedUser, String user
             throw new AuthenticationFailedException(errorMessage, e);
         }
     }
+
+    private String getFederatedUserId(AuthenticatedUser authenticatedUser, AuthenticationContext context,
+                                      String tenantDomain) throws TOTPException {
+
+        String userId;
+        String username = authenticatedUser.getUserName();
+        String userStoreDomain = authenticatedUser.getUserStoreDomain();
+        String idpName = authenticatedUser.getFederatedIdPName();
+        int tenantId = getTenantID(tenantDomain);
+        int appTenantId = IdentityTenantUtil.getTenantId(context.getTenantDomain());
+
+        try {
+            int idpId = UserSessionStore.getInstance().getIdPId(idpName, appTenantId);
+            userId = UserSessionStore.getInstance().
+                    getUserId(username, tenantId, userStoreDomain, idpId);
+            if (userId == null) {
```

Comment:
better to use StringUtils.isNotBlank here

Reply:
Fixed